verilog/CST/constraints_test.cc

```diff
@@ -0,0 +1,67 @@
+// Copyright 2017-2020 The Verible Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "verilog/CST/constraints.h"
+
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+#include "absl/strings/string_view.h"
+#include "common/analysis/matcher/matcher.h"
+#include "common/analysis/matcher/matcher_builders.h"
+#include "common/text/concrete_syntax_leaf.h"
+#include "common/text/concrete_syntax_tree.h"
+#include "common/text/symbol.h"
+#include "common/text/text_structure.h"
+#include "common/text/token_info.h"
+#include "common/util/casts.h"
+#include "common/util/logging.h"
+#include "verilog/CST/verilog_matchers.h"  // IWYU pragma: keep
+#include "verilog/CST/verilog_nonterminals.h"
+#include "verilog/analysis/verilog_analyzer.h"
+#include "verilog/parser/verilog_token_enum.h"
+
+#undef ASSERT_OK
+#define ASSERT_OK(value) ASSERT_TRUE((value).ok())
+
+namespace verilog {
+namespace {
+
+// Tests that GetSymbolIdentifierFromConstraintDeclaration correctly returns
+// the token of the symbol identifier.
+TEST(GetSymbolIdentifierFromConstraintDeclarationTest, BasicTests) {
+  const std::pair<std::string, absl::string_view> kTestCases[] = {
+      {"class foo; rand logic a; constraint Bar { a < 16; } endclass", "Bar"},
+      {"class foo; rand logic a; constraint b { a >= 16; } endclass", "b"},
+      {"class foo; rand logic a; constraint stH { a == 16; } endclass", "stH"},
+  };
+  for (const auto& test : kTestCases) {
+    VerilogAnalyzer analyzer(test.first, "");
+    ASSERT_OK(analyzer.Analyze());
+    const auto& root = analyzer.Data().SyntaxTree();
+    std::vector<verible::TreeSearchMatch> constraint_declarations =
+      verible::SearchSyntaxTree(*root, NodekConstraintDeclaration());
```

Comment:
Define a `FindAllConstraintDeclarations` for this, in .h and .cc?
Add a test that finds more than one constraint declaration.

Reply:
Sure, added such function + wrote some tests for it.